afs-core/src/test/java/com/powsybl/afs/AfsBaseTest.java

```diff
@@ -566,4 +563,40 @@ void hasDeepDependencyTest() {
         otherFile.setDependencies("dep", Collections.singletonList(createdFile));
         assertTrue(createdFile.hasDeepDependency(createdFile, "dep"));
     }
+
+    @Test
+    public void invalidate() {
+        Folder folder = afs.getRootFolder().createFolder("testFolder");
+        Project project = folder.createProject("test");
+        FooFile fooFile = project.getRootFolder().fileBuilder(FooFileBuilder.class).withName("Foo").build();
+        WithDependencyFile fileWithDep = project.getRootFolder().fileBuilder(WithDependencyFileBuilder.class).build();
+        fileWithDep.setFooDependency(fooFile);
+
+        // Previous code add connected WithDependencyFile with listeners
+        storage.getEventsBus().flush();
+        storage.getEventsBus().removeListeners();
+
+        fooFile.invalidate();
```

Comment:
is there a way to check that this line did in fact invalidate the ProjectNodes and that the parameter `false` in the invalidate method was effective?

Reply:
There is a new behavior to take this remark into account.
The description has been updated in the breaking changes section.